Modules/_asynciomodule.c

```diff
@@ -3241,6 +3241,9 @@ PyRunningLoopHolder_tp_dealloc(PyRunningLoopHolder *rl)
 {
     Py_CLEAR(rl->rl_loop);
     PyObject_Free(rl);
+    if (cached_running_holder == (PyObject *)rl) {
+        cached_running_holder = NULL;
+    }
```

Comment:
Can you move the code at the beginning of the function?

Reply:
sure, just fixed that up